src/views/Settings.vue

```diff
@@ -6,14 +6,22 @@
 <script lang="ts">
 import { defineComponent } from 'vue';
 import AddExercise from '@/components/AddExercise.vue';
-
+import * as API from '@/API';
 export default defineComponent({
+  async beforeMount() {
+    if (((await API.getRole()) as string) != 'admin') {

```

Comment:
typing, the non null assertions is not neccessary

Reply:
done